drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -5970,7 +5970,7 @@ Error RenderingDeviceVulkan::buffer_clear(RID p_buffer, uint32_t p_offset, uint3
 	return OK;
 }
 
-Vector<uint8_t> RenderingDeviceVulkan::buffer_get_data(RID p_buffer) {
+uint64_t RenderingDeviceVulkan::_buffer_get_data(RID p_buffer, uint8_t **p_mem, const std::function<void *(size_t)> &p_malloc) {
```

Comment:
Not really happy with this, I get the intention but its better you pass the pointer to where to write and dont allocate. You are supposed to know the size of this, so returning it is not necesary. Feel free to add a buffer_get_size if you want to query it.